### PR TITLE
fixing DifferentiableGraphOp updating requires_grad on input tensor list; python test added to verify the test

### PR DESCRIPTION
**Pull request by jjsjann123**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #57575 Rework requires_grad on DifferentiableGraphOp
* **#57574 fixing DifferentiableGraphOp updating requires_grad on input tensor list; python test added to verify the test**

Differential Revision: [D29038774](https://our.internmc.facebook.com/intern/diff/D29038774)

### CODE REVIEW

**Comment by jjsjann123** [comment]
This is the bug fix we have in #56466 that was asked to be put into a separate PR.
@Krovatkin

**Comment by jjsjann123** [comment]
Bumping to see if there's any update

**Review by Krovatkin — APPROVED** [review]
:shipit:

File: torch/csrc/jit/runtime/graph_executor.cpp [hunk 1]
```diff
@@ -439,9 +439,9 @@ struct DifferentiableGraphOp {
     if (v.isTensor()) {
       v = IValue(detach(std::move(v).toTensor()));
     } else if (v.isTensorList()) {
-      c10::List<at::Tensor> lst = std::move(v).toTensorList();
+      std::vector<at::Tensor> lst = v.toTensorVector();
       for (size_t i = 0; i < lst.size(); ++i) {
```
Comment (Krovatkin) [inline torch/csrc/jit/runtime/graph_executor.cpp:443]: could you stick no-clang-tidy in here or change it to a range-based to keep clang-tidy happy?

**Comment by Krovatkin** [comment]
@Krovatkin has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D29038774).

**Pull request closed without merging**